finality-aleph/src/network/manager/discovery.rs

```diff
@@ -31,38 +32,51 @@ impl DiscoveryMessage {
 }
 
 /// Handles creating and responding to discovery messages.
-pub struct Discovery {
+pub struct Discovery<M: Multiaddress> {
     cooldown: Duration,
     last_broadcast: HashMap<NodeIndex, Instant>,
+    _phantom: PhantomData<M>,
 }
 
-type DiscoveryCommand = (DiscoveryMessage, DataCommand);
+type DiscoveryCommand<M> = (
```

Comment:
@fbielejec mentions it below but I'm generally against using tuples for anything other than trivial unpacking of values or enumeration in lists.
```suggestion
struct DiscoveryCommand<M> {
  message: DiscoverMessage<M>,
  command: DataCommand<<M as Multiaddress>::PeerId>,
}
```

Reply:
As discussed elsewhere we will leave this for now and try agreeing on a policy on tuples vs structs later. I sent you the task.